woocommerce/woocommerce-android
# Order Creation: change product quantities

Closes: #5514 


### Description
This is a small PR to handle quantity changes, I left a TODO comment about any potential controlling of the maximum quantity, as we need to discuss it with the whole team, since Core doesn't do it.

### Testing instructions
Same testing instructions as #5512, with the patch below, then change the quantities using the stepper buttons, and confirm they work as expected.

### Patch
```
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationViewModel.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationViewModel.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationViewModel.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationViewModel.kt	(revision 9ae0a0d0cf7ee3deac9d07bf106ecc10d3ddcf04)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationViewModel.kt	(date 1640276818412)
@@ -16,6 +16,7 @@
 import com.woocommerce.android.viewmodel.ScopedViewModel
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.withContext
+import java.math.BigDecimal
 import javax.inject.Inject
 
 @HiltViewModel
@@ -57,6 +58,49 @@
         orderDraft = orderDraft.copy(
             currency = parameterRepository.getParameters(PARAMETERS_KEY, savedState).currencySymbol.orEmpty()
         )
+        orderDraft = orderDraft.copy(
+            items = listOf(
+                Order.Item(
+                    0L,
+                    productId = 44,
+                    "Test Product",
+                    price = BigDecimal.TEN,
+                    sku = "SKU123",
+                    quantity = 2f,
+                    subtotal = BigDecimal.ONE,
+                    totalTax = BigDecimal.TEN,
+                    total = BigDecimal.TEN,
+                    variationId = 0L,
+                    attributesList = emptyList()
+                ),
+                Order.Item(
+                    0L,
+                    productId = 45,
+                    "Test Product",
+                    price = BigDecimal.TEN,
+                    sku = "SKU124",
+                    quantity = 2f,
+                    subtotal = BigDecimal.ONE,
+                    totalTax = BigDecimal.TEN,
+                    total = BigDecimal.TEN,
+                    variationId = 0L,
+                    attributesList = emptyList()
+                ),
+                Order.Item(
+                    0L,
+                    productId = 47,
+                    "Test Product",
+                    price = BigDecimal.TEN,
+                    sku = "SKU125",
+                    quantity = 2f,
+                    subtotal = BigDecimal.ONE,
+                    totalTax = BigDecimal.TEN,
+                    total = BigDecimal.TEN,
+                    variationId = 0L,
+                    attributesList = emptyList()
+                )
+            )
+        )
     }
 
     fun onOrderStatusChanged(status: Order.Status) {
@@ -80,7 +124,7 @@
         }
         orderDraft = orderDraft.copy(items = items)
     }
-
+    
     private suspend fun Order.Item.toProductUIModel(): ProductUIModel {
         val (isStockManaged, stockQuantity) = withContext(dispatchers.io) {
             if (isVariation) {
```
